src/dll/ScriptCompiler.hpp

```diff
@@ -0,0 +1,91 @@
+#pragma once
+#include <scc.h>
+
+class ScriptCompilerSourceRef
+{
+public:
+    uint8_t GetType() const;
+    std::string_view GetName() const;
+    std::string_view GetParentName() const;
+    std::string_view GetPath() const;
+    size_t GetLine() const;
+    bool IsNative() const;
+
+private:
+    ScriptCompilerSourceRef(SccApi& api, SccOutput* output, SccSourceRef* sourceRef);
+
+    SccApi& m_scc;
+    SccOutput* m_output;
+    SccSourceRef* m_sourceRef;
+
+    friend class ScriptCompilerOutput;
+};
+
+class ScriptCompilerOutput
+{
+    // Key used to prevent construction of this class outside of ScriptCompilerSettings.
+    // This is necessary to construct the resulting std::variant in place.
+    struct Key
+    {
+        explicit Key() = default;
+    };
+
+public:
+    ScriptCompilerOutput(SccApi& scc, SccResult* result, SccOutput* output, Key key);
+    ScriptCompilerOutput(const ScriptCompilerOutput&) = delete;
+    ScriptCompilerOutput& operator=(const ScriptCompilerOutput&) = delete;
+    ~ScriptCompilerOutput();
+
+    size_t GetSourceRefCount() const;
+    ScriptCompilerSourceRef GetSourceRef(size_t index) const;
+
+private:
+    SccApi& m_scc;
+    SccResult* m_result;
+    SccOutput* m_output;
+
+    friend class ScriptCompilerSettings;
+};
+
+class ScriptCompilerFailure
+{
+    // Key used to prevent construction of this class outside of ScriptCompilerSettings.
+    // This is necessary to construct the resulting std::variant in place.
+    struct Key
+    {
+        explicit Key() = default;
+    };
+
+public:
+    ScriptCompilerFailure(SccApi& scc, SccResult* result, Key key);
+    ScriptCompilerFailure(const ScriptCompilerFailure&) = delete;
+    ScriptCompilerFailure& operator=(const ScriptCompilerFailure&) = delete;
+    ~ScriptCompilerFailure();
+
+    const std::string& GetMessage() const;
+
+private:
+    SccApi& m_scc;
+    SccResult* m_result;
+    std::string m_message;
+
+    friend class ScriptCompilerSettings;
+};
+
+class ScriptCompilerSettings
+{
+public:
+    using Result = std::variant<ScriptCompilerFailure, ScriptCompilerOutput>;
+
+    ScriptCompilerSettings(SccApi& m_scc, std::u8string r6Path);
```

Comment:
```suggestion
    ScriptCompilerSettings(SccApi& m_scc, std::filesystem::path r6Dir);
```

I would say to use `std::filesystem::path` when a path is needed instead of a u8string and when calling the API it should be converted to the string.